Sources/WhisperKit/Core/WhisperKit.swift

```diff
@@ -282,35 +308,46 @@ open class WhisperKit {
             Logging.debug("Loaded feature extractor")
         }
 
-        if let audioEncoder = audioEncoder as? WhisperMLModel {
-            Logging.debug("Loading audio encoder")
-            try await audioEncoder.loadModel(
-                at: encoderUrl,
-                computeUnits: modelCompute.audioEncoderCompute,
+        if FileManager.default.fileExists(atPath: decoderPrefillUrl.path) {
+            Logging.debug("Loading text decoder prefill data")
+            textDecoder.prefillData = TextDecoderContextPrefill()
+            try await textDecoder.prefillData?.loadModel(
+                at: decoderPrefillUrl,
+                computeUnits: modelCompute.prefillCompute,
                 prewarmMode: prewarmMode
             )
-            Logging.debug("Loaded audio encoder")
+            Logging.debug("Loaded text decoder prefill data")
         }
 
         if let textDecoder = textDecoder as? WhisperMLModel {
             Logging.debug("Loading text decoder")
+            let decoderLoadStart = modelLoadStart //CFAbsoluteTimeGetCurrent()
             try await textDecoder.loadModel(
                 at: decoderUrl,
                 computeUnits: modelCompute.textDecoderCompute,
                 prewarmMode: prewarmMode
             )
+            let decoderLoadEnd = CFAbsoluteTimeGetCurrent()
+
+            currentTimings.decoderLoadTime = (decoderLoadEnd - decoderLoadStart)
+
             Logging.debug("Loaded text decoder")
         }
 
-        if FileManager.default.fileExists(atPath: decoderPrefillUrl.path) {
-            Logging.debug("Loading text decoder prefill data")
-            textDecoder.prefillData = TextDecoderContextPrefill()
-            try await textDecoder.prefillData?.loadModel(
-                at: decoderPrefillUrl,
-                computeUnits: modelCompute.prefillCompute,
+        if let audioEncoder = audioEncoder as? WhisperMLModel {
+            Logging.debug("Loading audio encoder")
+            let encoderLoadStart = modelLoadStart // CFAbsoluteTimeGetCurrent()
```

Comment:
Curious about this part, is CFAbsoluteTimeGetCurrent() having some accuracy issue?